sql/src/main/java/io/crate/execution/ddl/RerouteActions.java

```diff
@@ -86,6 +93,27 @@ private RerouteActions() {
         return listener;
     }
 
+    public static ClusterRerouteRequest prepareAllocateStalePrimaryReq(PromoteReplicaStatement promoteReplica,
+                                                                       Functions functions,
+                                                                       Row parameters,
+                                                                       TransactionContext txnCtx) {
+        String index = RerouteActions.getRerouteIndex(promoteReplica, parameters);
+        Integer shardId = DataTypes.INTEGER.value(SymbolEvaluator.evaluate(
+            txnCtx, functions, promoteReplica.shardId(), parameters, SubQueryResults.EMPTY));
+        if (shardId == null) {
+            throw new NullPointerException("Shard in REROUTE PROMOTE REPLICA must not be null");
+        }
+        Boolean acceptDataLoss = DataTypes.BOOLEAN.value(SymbolEvaluator.evaluate(
+            txnCtx, functions, promoteReplica.acceptDataLoss(), parameters, SubQueryResults.EMPTY));
+        if (acceptDataLoss == null) {
```

Comment:
if `accept_data_loss` is false, an exception containing the `index` is raised. this may confuse users as index is internally. I guess this applies to all reroute commands, maybe we should solve this for all as a follow up.

Reply:
Good point, will look into this later.